### ISSUE TITLE
MaterialCardView crashes when including a layout

### ISSUE THREAD

**Issue by pauliusrum** [material-components/material-components-android#419]
Hi,

I have two layouts:
- main.xml
```xml
<layout xmlns:android="http://schemas.android.com/apk/res/android">

    <com.google.android.material.card.MaterialCardView
        android:layout_width="match_parent"
        android:layout_height="wrap_content">

        <include layout="@layout/child"/>

    </com.google.android.material.card.MaterialCardView>
</layout>
```
- child.xml
```xml
<layout xmlns:android="http://schemas.android.com/apk/res/android">

    <androidx.constraintlayout.widget.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent">

    </androidx.constraintlayout.widget.ConstraintLayout>
</layout>
```

When I include `child.xml` in `main.xml` the app crashes with this error:
```
2019-07-07 21:40:20.293 9737-9737/com.tasker E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tasker, PID: 9737
    java.lang.NullPointerException: Attempt to invoke virtual method 'void main.databinding.ChildBinding.invalidateAll()' on a null object reference
        at main.databinding.MainBindingImpl.invalidateAll(MainBindingImpl.java:52)
        at main.databinding.MainBindingImpl.<init>(MainBindingImpl.java:44)
        at main.databinding.MainBindingImpl.<init>(MainBindingImpl.java:33)
        at main.DataBinderMapperImpl.getDataBinder(DataBinderMapperImpl.java:118)
        at androidx.databinding.MergedDataBinderMapper.getDataBinder(MergedDataBinderMapper.java:74)
        at androidx.databinding.DataBindingUtil.bind(DataBindingUtil.java:199)
        at androidx.databinding.DataBindingUtil.inflate(DataBindingUtil.java:130)
        at androidx.databinding.ViewDataBinding.inflateInternal(ViewDataBinding.java:1366)
        at main.databinding.MainBinding.inflate(MainBinding.java:35)
        at main.databinding.MainBinding.inflate(MainBinding.java:21)
        at schedule.FlightListAdapter.onCreateViewHolder(MyFlightsFragment.kt:158)
        at schedule.FlightListAdapter.onCreateViewHolder(MyFlightsFragment.kt:120)
        at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6957)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6126)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6009)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6005)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2302)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1627)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4040)
        at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3446)
        at android.view.View.measure(View.java:24966)
        at androidx.constraintlayout.widget.ConstraintLayout.internalMeasureChildren(ConstraintLayout.java:1227)
        at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:1572)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:24966)
        at androidx.constraintlayout.widget.ConstraintLayout.internalMeasureChildren(ConstraintLayout.java:1227)
        at androidx.constraintlayout.widget.ConstraintLayout.onMeasure(ConstraintLayout.java:1572)
        at android.view.View.measure(View.java:24966)
        at androidx.drawerlayout.widget.DrawerLayout.onMeasure(DrawerLayout.java:1119)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1535)
        at android.widget.LinearLayout.measureVertical(LinearLayout.java:825)
        at android.widget.LinearLayout.onMeasure(LinearLayout.java:704)
2019-07-07 21:40:20.305 9737-9737/com.tasker E/AndroidRuntime:     at android.view.View.measure(View.java:24966)
        at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7139)
        at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
        at com.android.internal.policy.DecorView.onMeasure(DecorView.java:992)
        at android.view.View.measure(View.java:24966)
        at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:3278)
        at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:2005)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2307)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1865)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8488)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
        at android.view.Choreographer.doCallbacks(Choreographer.java:761)
        at android.view.Choreographer.doFrame(Choreographer.java:696)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7045)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```

If I replace `com.google.android.material.card.MaterialCardView` with `androidx.cardview.widget.CardView` it stops crashing.

**Comment by ymarian** [comment]
@pauliusrum Seems similar to https://issuetracker.google.com/issues/135272556 can you see if there is a similar workaround or share a sample project so I can try to find one?

We've been trying to remove the extra view, but there is still some work to do for that.

**Comment by pauliusrum** [comment]
The problem is the same, but I'm inflating the layout in a `RecyclerView` so that workaround does not work. 
I have found a workaround to move `ConstraintLayout` outside of the include statement but this will not do as a long term solution.  e.g.
```xml
<!-- Should be inside of @layout/child -->
<androidx.constraintlayout.widget.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="wrap_content">
    <include layout="@layout/child"/>
</androidx.constraintlayout.widget.ConstraintLayout>        
```
The issue status in google tracker is Won't Fix (Intended behavior) so what are the other options? 
I could provide an example project but it will take some time.

**Comment by ymarian** [comment]
Long term you can follow updates to:
http://issuetracker.google.com/b/122304308

I'm convinced we should do it now, it's just blocked on a few changes that need to be made before.
